syntax = "proto3";

option go_package = "/protobufs";

service RunModelService {
    rpc RunModel (RunModelRequest) returns (RunModelResponse);
}
message FinalData {
    message WeatherMetrics {
        double avg_temperature = 1;
        double temp_std_dev = 2;
        double avg_humidity = 3;
        double humidity_std_dev = 4;
        double total_precipitation = 5;
        int32 dry_days_consecutive = 6;
    }
    WeatherMetrics weather = 1;
    message DeltaData {
        string forest = 1;
        string plot = 2;
        int32 delta_min = 3;
        int32 delta_max = 4;
        int32 delta = 5;
        string start_date = 6;
        string end_date = 7;
        int32 x = 8;
        int32 y = 9;
        double ndre = 10;
        double ndmi = 11;
        double psri = 12;
        double ndvi = 13;
        double ndre_derivative = 14;
        double ndmi_derivative = 15;
        double psri_derivative = 16;
        double ndvi_derivative = 17;
        string label = 18;
        double latitude = 19;
        double longitude = 20;
    }
    DeltaData delta = 2;
    string created_at = 3;
}

message PixelResult {
    int32 x = 1;
    int32 y = 2;
    double latitude = 3;
    double longitude = 4;
    repeated LabelProbability result = 5;
}

message LabelProbability {
    string label = 1;
    double probability = 2;
}

message RunModelRequest {
    repeated FinalData data = 1;
    string model = 2;
}

message RunModelResponse {
    repeated PixelResult results = 1;
}